engine/classes/Elgg/I18n/Translator.php

```diff
@@ -299,6 +299,11 @@ public function loadTranslations($language = null) {
 				$this->loaded_from_cache = true;
 				$this->registerLanguagePath($this->defaultPath);
 				$this->is_initialized = true;
+
+				elgg_register_event_handler('cache:flush', 'system', function() {
```

Comment:
I am not sure about this though. Does flushing caches rebuild translations from cache?

Reply:
it probably couldn't because the cached files are removed

Reply:
So, my thinking is right here?